auto-value-cursor/src/main/java/com/gabrielittner/auto/value/cursor/AutoValueCursorExtension.java

```diff
@@ -221,6 +223,53 @@ private static boolean hasAnnotationWithName(Element element, String simpleName)
         return false;
     }
 
+    private static AnnotationMirror getAnnotationMirror(Element element, Class<?> clazz) {
+        String clazzName = clazz.getName();
+        for (AnnotationMirror m : element.getAnnotationMirrors()) {
+            if (m.getAnnotationType().toString().equals(clazzName)) {
+                return m;
+            }
+        }
+        return null;
+    }
+
+    private static AnnotationValue getAnnotationValue(AnnotationMirror annotationMirror, String key) {
+        for (Entry<? extends ExecutableElement, ? extends AnnotationValue> entry :
+                annotationMirror.getElementValues().entrySet()) {
+            if (entry.getKey().getSimpleName().toString().equals(key)) {
+                return entry.getValue();
+            }
+        }
+        return null;
+    }
+
+    public TypeMirror getFactoryTypeMirror(Element element) {
+        AnnotationMirror annotationMirror =
+                getAnnotationMirror(element, CursorAdapter.class);
+        if (annotationMirror == null) {
+            return null;
+        }
+        AnnotationValue annotationValue = getAnnotationValue(annotationMirror, "value");
+        return annotationValue == null ? null : (TypeMirror) annotationValue.getValue();
+    }
+
+    private String getFactoryMethodName(TypeName returnType, TypeElement factoryClass) {
+        List<? extends Element> elements = factoryClass.getEnclosedElements();
+        for (Element element : elements) {
+            // searching for a static method
+            if (element.getKind() != ElementKind.METHOD
+                    || !element.getModifiers().contains(Modifier.STATIC)) {
+                continue;
+            }
+            ExecutableElement method = (ExecutableElement) element;
+            // that method should return the annotated class and take a Cursor as parameter
+            if (methodTakesAndReturns(method, CURSOR, returnType)) {
+                return method.getSimpleName().toString();
+            }
+        }
+        return null;
```

Comment:
Throw here with an appropriate error message instead of returning null. Right now javapoet would throw an IllegalArgumentException in line 132 for passing in null.